pkg/autodiscovery/listeners/kube_services.go

```diff
@@ -204,6 +203,19 @@ func (l *KubeServiceListener) createService(ksvc *v1.Service, firstRun bool) {
 	if ksvc == nil {
 		return
 	}
+
+	l.m.Lock()
+	l.endpointsAnnotated[fmt.Sprintf("%s/%s", ksvc.Namespace, ksvc.Name)] = isEndpointAnnotated(ksvc)
+	endpoints, err := l.endpointsInformer.Lister().List(labels.Everything())
```

Comment:
Calling this concrete method will make it harder to test `createService` itself, we could keep a new `l.endpointsLister` of type [`EndpointsLister` (interface)](https://godoc.org/k8s.io/client-go/listers/core/v1#EndpointsLister), and during tests change it to a mock implementing the `EndpointsLister` and `EndpointsNamespaceLister` interfaces.

See `mockedPluggableAutoConfig` and `pluggableAutoConfig` interface in the clusterchecks package